csi/controllerserver.py

```diff
@@ -289,6 +289,24 @@ def CreateVolume(self, request, context):
                 context.set_code(grpc.StatusCode.RESOURCE_EXHAUSTED)
                 return csi_pb2.CreateVolumeResponse()
 
+        if filters.get('kadalu_format') == 'non-native':
+            # Then mount the whole volume as PV
+            msg = "non-native way of Kadalu mount expected"
+            logging.info(msg)
+            return csi_pb2.CreateVolumeResponse(
+                volume={
+                    "volume_id": request.name,
+                    "capacity_bytes": pvsize,
+                    "volume_context": {
+                        "type": hostvoltype,
+                        "hostvol": hostvol,
+                        "pvtype": pvtype,
+                        "fstype": "xfs",
+                        "kadalu_format": "non-native"
```

Comment:
I think it's a good idea to use this field in other responses (3 other places in same fn) as well, just to be consistent?

Reply:
Done